xray/conventions.py

```diff
@@ -64,11 +64,17 @@ def mask_and_scale(array, fill_value=None, scale_factor=None, add_offset=None,
 
 
 def _netcdf_to_numpy_timeunit(units):
+    units = units.lower()
+    if not units.endswith('s'):
+        units = '%ss' % units
     return {'seconds': 's', 'minutes': 'm', 'hours': 'h', 'days': 'D'}[units]
 
 
 def _unpack_netcdf_time_units(units):
-    matches = re.match('(\S+) since (.+)', units).groups()
+    # CF datetime units follow the format: "UNIT since DATE"
+    # this parses out the unit and date allowing for extraneous
+    # whitespace.
+    matches = re.match('\s*(\S+)\s+since\s+(.+)\s*', units).groups()
```

Comment:
How about using `units.strip()` instead of adding the patterns to match trailing/starting whitespace?

This version doesn't trim the whitespace the end:
```
In [1]: import re

In [2]: units = ' Hour  since  1680-01-01 00:00:00 '

In [3]: re.match('\s*(\S+)\s+since\s+(.+)\s*', units).groups()
Out[3]: ('Hour', '1680-01-01 00:00:00 ')
```